src/dndbeyond/content-scripts/character.js

```diff
@@ -857,24 +857,97 @@ function displayAction(paneClass) {
+    if (matchedText==undefined || matchedText==null) {
+        msgElement = pane.find(".ct-action-detail__description").get(0);
+        console.log(msgElement);
+        if (msgElement!=undefined) {
+            elemText = msgElement.innerText;
+            // console.log("Description Text: "+elemText);    
+            matchedText = [...elemText.matchAll(regexp)]; 
+        }
+    }
+
+    //TODO possibly add the ability to have the note text add its own, separate roll button, instead of using the existing ones
+    //Handle Matched Text
+    if (matchedText!=undefined && matchedText!=null) {
+        console.log("Matched Text:\n"+matchedText);
+        for (let i=0; i<matchedText.length; i++) {
+            let cMatchType = matchedText[i][1].toLowerCase();
+            let cMatchText = matchedText[i][2];
+            // Determine if we should prevent the default action/message from being sent to the VTT
+            console.log("Note Match #"+i+" (Provided Type: "+cMatchType+"):\n"+cMatchText);
+            if (rollOrderTypes.includes(cMatchType))
+                rollOrder = cMatchType;
+            rollFunc = function () { 
+                sendRollWithCharacter("action", 0, {
+                    "type": "raw-text",
+                    "text": cMatchText
+                })
+            };
+        }
+    }
+    
+    return { rollOrder: rollOrder, rollFunction: rollFunc };
+}
+
 function execute(paneClass) {
     console.log("Beyond20: Executing panel : " + paneClass);
-    if (["ct-skill-pane", "ct-custom-skill-pane"].includes(paneClass))
-        rollSkillCheck(paneClass);
-    else if (paneClass == "ct-ability-pane")
-        rollAbilityCheck();
-    else if (paneClass == "ct-ability-saving-throws-pane")
-        rollSavingThrow();
-    else if (paneClass == "ct-initiative-pane")
-        rollInitiative();
-    else if (paneClass == "ct-item-pane")
-        rollItem();
-    else if (["ct-action-pane", "ct-custom-action-pane"].includes(paneClass))
-        rollAction(paneClass);
-    else if (paneClass == "ct-spell-pane")
-        rollSpell();
-    else
-        displayPanel(paneClass);
+    const execRolls = function() {
+        if (["ct-skill-pane", "ct-custom-skill-pane"].includes(paneClass))
+            rollSkillCheck(paneClass);
+        else if (paneClass == "ct-ability-pane")
+            rollAbilityCheck();
+        else if (paneClass == "ct-ability-saving-throws-pane")
+            rollSavingThrow();
+        else if (paneClass == "ct-initiative-pane")
+            rollInitiative();
+        else if (paneClass == "ct-item-pane")
+            rollItem();
+        else if (["ct-action-pane", "ct-custom-action-pane"].includes(paneClass))
+            rollAction(paneClass);
+        else if (paneClass == "ct-spell-pane")
+            rollSpell();
+        else
+            displayPanel(paneClass);  
+    }
+
+    //TODO allow adding additional custom roll modifiers within custom text?
```

Comment:
It might make more sense to have the execute just pass along the callback to handleCustomText, but I'm not sure about the design here, something seems off to me.